publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java

```diff
@@ -483,13 +510,14 @@ public void shutdownCompleted(ShutdownSignalException cause) {
             throw new TimeoutException("Timeout waiting for ACK " + e.getMessage());
         } catch (IllegalArgumentException e) {
             log.error("Failed to publish message due to " + e.getMessage());
-            throw new IllegalArgumentException("DomainId limit exceeded " + e.getMessage());
+            throw new IllegalArgumentException("DomainId limit exceeded " + e.getMessage(), e);
         } catch (Exception e) {
             log.error(e.getMessage(), e);
             if(!channel.isOpen()&& rabbitConnection.isOpen()){
-                throw new RemRemPublishException("Channel was closed for Rabbitmq connection ::" + factory.getHost() + factory.getPort());
+                throw new RemRemPublishException("Channel was closed for Rabbitmq connection ::",
+                        factory, e);
             }
-            throw new IOException("Failed to publish message due to " + e.getMessage());
+            throw new IOException("Failed to publish message due to " + e.getMessage(), e);
```

Comment:
An old error corrected or a mistake?

Reply:
It was suggested to pass an original exception when another exception is thrown

Reply:
Yes, it's good practice to chain exception, so we can get stack-trace of original exception and find where an exception originates from.